pyanaconda/modules/boss/module_manager/start_modules.py

```diff
@@ -75,39 +78,41 @@ def run(self):
 
         return self._module_observers
 
-    def _find_modules(self):
-        """Find modules."""
-        modules = []
+    @staticmethod
+    def _match_module(name, patterns):
+        """Match a module with one of the specified patterns."""
+        for pattern in patterns:
+            # Match the name prefix.
+            if pattern.endswith("*") and name.startswith(pattern[:-1]):
+                return True
 
-        for service_name in self._module_names:
-            log.debug("Found %s.", service_name)
-            modules.append(ModuleObserver(
-                self._message_bus,
-                service_name
-            ))
+            # Match the full name.
+            if name == pattern:
+                return True
 
-        return modules
+        return False
 
-    def _find_addons(self):
-        """Find additional modules."""
+    def _find_modules(self):
+        """Find modules to start."""
         modules = []
 
-        if not self._addons_enabled:
-            return modules
-
         dbus = self._message_bus.proxy
         names = dbus.ListActivatableNames()
-        prefix = get_dbus_name(*ADDONS_NAMESPACE)
 
         for service_name in names:
-            if not service_name.startswith(prefix):
+            # Only activitable modules can be started.
+            if not self._match_module(service_name, self._activitable):
+                continue
+
+            # Forbidden modules are not allowed to run.
+            if self._match_module(service_name, self._forbidden):
+                log.debug("Skip %s. The module is forbidden.", service_name)
```

Comment:
Maybe this could be improved - suggestion:
"Module X will not be started, because it's marked as forbidden in the Anaconda profile file." or similar ?

Basically I think we should say what is skipped (an Anaconda DBus module) and why/where to change that (in the Anaconda config file).